### PR TITLE
fix races

### PR DESCRIPTION
**Pull request by ngauthier**
@bfosberry 

Also I am unwrapping the 1 error condition. Doesn't seem to break anything on jet.

### CODE REVIEW

File: background_worker.go [hunk 1]
```diff
@@ -50,22 +50,26 @@ func (b *backgroundWorker) Close() ([]interface{}, error) {
...
+
+	done := make(chan bool)
 	go func() {
-		for {
-			valueError, ok := <-b.valueErrors
-			if !ok {
-				break
-			}
-			values = append(values, valueError.value)
+		for valueError := range b.valueErrors {
```
Comment (bfosberry) [inline background_worker.go]: YES

File: background_worker.go [hunk 2]
```diff
@@ -50,22 +50,26 @@ func (b *backgroundWorker) Close() ([]interface{}, error) {
 	if err := b.destroyable.Destroy(); err != nil {
 		errs = append(errs, err)
 	}
+
+	done := make(chan bool)
 	go func() {
```
Comment (bfosberry) [inline background_worker.go]: TBH I would switch these around. Have the main flow blocked by pulling from valueErrors, and start a new goroutine to wait for the waitgroup and then close the channel

**Comment by bfosberry** [comment]
:+1:

File: background_worker.go (hunk 2 above)
Comment (ngauthier) [inline background_worker.go]: TBH my next change will be `rm -rf` :D

**Pull request merged**